src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BasicScenarioTests.cs

```diff
@@ -40,5 +42,31 @@ public static IEnumerable<TestScenario> GetScenarios()
             yield return new(nameof(BasicScenarioTests), language, DotNetTemplate.NUnit,    DotNetActions.Test);
             yield return new(nameof(BasicScenarioTests), language, DotNetTemplate.MSTest,   DotNetActions.Test);
         }
+
+        yield return new(nameof(BasicScenarioTests), DotNetLanguage.CSharp, DotNetTemplate.WebApp, DotNetActions.PublishSelfContained, VerifyRuntimePacksForSelfContained);
+    }
+
+    private static void VerifyRuntimePacksForSelfContained(string projectPath)
+    {   
+        string projNugetCachePath = Path.Combine(projectPath, "obj", "project.nuget.cache");
+        JsonNode? projNugetCache = JsonNode.Parse(File.ReadAllText(projNugetCachePath));
+
+        string failMessage = "Runtime packs were retrieved from NuGet instead of the SDK";
+        string? restoredPackageFiles = projNugetCache?["expectedPackageFiles"]?.ToString();
+
+        if ( restoredPackageFiles != null)
+        {
+            Assert.True("[]" == restoredPackageFiles, failMessage);
+        } else
+        {
+            throw new NugetCacheParseException();
```

Comment:
I don't think this warrants a new exception type.  The primary reason is that nobody needs a specific type to catch.  Additionally instead of incurring the overhead of throwing an exception, consider utilizing an Assert.